### DISCUSSION TITLE
Zoom not working

### DISCUSSION THREAD

**Discussion by dirri** [GSI-CS-CO/chart-fx#524] [Q&A]
When i zoom in by drawing a rectangle via LMB over a certain area of the chart the result is an empty canvas. This seems to be sporadically. Also when using the mouse wheel result sometime to empty canvas. I did not found out yet what the reason is!?
Example:
![grafik](https://user-images.githubusercontent.com/11440450/170944034-eb8c993b-ce71-4902-bf36-bea70ef2943e.png)
Results to:
![grafik](https://user-images.githubusercontent.com/11440450/170944219-72851dd2-e1c8-422d-bc63-319debf5f1ab.png)
I used latest version of chartfx but this appears also with the JDK8-SNAPSHOT version.
Please help!

**Comment by dirri** [comment]
https://user-images.githubusercontent.com/11440450/171038711-b69e5a15-9203-400a-b765-96fd16fc1261.mp4

**Comment by wirew0rm** [comment]
Sorry for the late reply, if you can reproduce the problems in the current (jdk11) version in one of the samples or with a minimal reproducer that would make it easier to look into it.

With only screenshots it is very hard to figure out what is going wrong here, since there are so many different ways to set up a chart in chart-fx. So please share a minified version of your code if you cannot reproduce it in the samples and also describe what you are doing different from the samples if the problem doesn't occur there.

Other things that are always worth trying is to disable data reduction in the renderer and/or make the renderer not assume sorted data. The type of dataset is also important, since it provides the lookup to get the data-points for a specific zoom range.

**Reply by RalphSteinhagen** [reply]
@dirri Chart-FX received some major upgrade recently. Could you verify against main if your problem still persists? Thanks in advamce.

**Comment by rponnuru1866** [comment]
Hi, I am facing a similar issue. I am using jdk17. I have attached the screenshots for reference. The problem only occurs when the Y-axis values are greater than 10 digits and if it is the left most datapoint in the graph. This occurs only in specific input data. If I zoom on multiple datapoints, it works fine but when I zoom specifically on the left most bar it disappears.

<img width="1114" alt="Screen Shot 2022-06-21 at 12 23 41 PM" src="https://user-images.githubusercontent.com/107945230/174853138-862506c6-15e0-47d4-8b84-f3f3cecd0329.png">

<img width="1114" alt="Screen Shot 2022-06-21 at 12 23 54 PM" src="https://user-images.githubusercontent.com/107945230/174853207-b84a3f82-a17e-460e-bcb7-4f5b612f4e8d.png">

<img width="1113" alt="Screen Shot 2022-06-21 at 12 24 22 PM" src="https://user-images.githubusercontent.com/107945230/174853230-fd641818-361c-48cb-8127-ae359455bd39.png">

**Reply by wirew0rm** [reply]
Can you share a minimal example which reproduces the bug (or even better reproduce it in one of the samples)? From images and the description alone I can only recommend the same generic things like disabling data reduction. There is complicated interplay between Axis, Renderer and Dataset which is a lot easier to sort out with a running sample.
Since you mention jdk17 specifically, didi it work for you on a different jdk or is that just the jdk you happen to be using?

**Reply by rponnuru1866** [reply]
Here's a link to a minimal example that reproduces the bug. 
https://drive.google.com/file/d/1WNAooO6wQsnecI1ftC1se_eFLSarpRrt/view?usp=sharing

I didn't try on other jdk versions, jdk17 is the one that I happen to be using.

**Reply by wirew0rm** [reply]
Hey, thanks for providing some code. According to the pom.xml you are using JDK18, not 17? I changed it to 17 because I had that one at hand. I would argue that the sample is not really minimal though. There are multiple levels of inheritance between the sample and chart-fx and lots of properties are set that I would suspect have nothing to do with the problem. I would suggest that you remove everything from that sample which is not necessary to reproduce the issue. Also you are not even using our Zoomer implementation but a heavily customized version of it if I understood the sample correctly. Does the issue also occur with the default Zoomer implementation?

The one thing I noticed is that the chart does not resize to the window size and if you drag outside of the chart area, a zoom is triggered anyway which should not happen. Also if you zoom in on almost the full chart area, you can see that it zooms to the upper half of the data. I suspect that the coordinate transform between the different panes is somehow incorrect.

**Reply by rponnuru1866** [reply]
Hi, here's the link to a more stripped down version of the example that I previously provided. Also in the new example provided, I am using the default zoomer plugin from your package.
https://drive.google.com/file/d/1YrgeNp8mXkzPKlhTRCvGnJVCUS_nepfH/view?usp=sharing

**Reply by RalphSteinhagen** [reply]
@rponnuru1866 Chart-FX received some major upgrade recently. Could you verify against main if your problem still persists? Thanks in advamce.

**Comment by dirri** [comment]
>  @dirri Chart-FX received some major upgrade recently. Could you verify against main if your problem still persists? Thanks in advamce. 

Which Version should i try ?

**Reply by wirew0rm** [reply]
These changes are not yet released, as there are still some followup changes that we are want to get in and fix before the release. Also keep in mind that this is a major update and includes some small code changes, most notably changing the package names from `de.gsi.chart` to `io.fair_acc.chartfx`.

To test the Snapshot version follow the instructions in the [README](https://github.com/milo-gsi/chart-fx/blob/main/README.md#using-the-snapshot-repository). This should be released in the coming weeks and feedback (related to the zoom issue or otherwise) on changes and the upgrading process would be most welcome.

**Comment by alexwenzelde** [comment]
Hi all,
I have the same issue when zooming into the chart. The data disappears. When I pan the chart, the chart comes back into sight, and at a certain point it disappears again. I created a minimal code sample with my data in a CSV file. I attached the project. Please run the App.java and press the "Plot Data" Button. 
If you open the file "PrimaryController.java" you can see how I created the chart. I already disabled the DataReduction as you mentioned in the previous posts, and the chart is resizable with the windows. I inserted your sinus-wave code sample. You can uncomment it in the primary controller, and as you can see, it works as it should.
I use openjfx 21.0.5 and the java jdk 22 (openjdk). I created the GUI with SceneBuilder and I use NetBeans.

I hope you can help me!

Regards, Alex

[ChartFxSample.zip](https://github.com/user-attachments/files/17956580/ChartFxSample.zip)

**Reply by alexwenzelde** [reply]
There is no need for help! I found the fix in this issue:

https://github.com/fair-acc/chart-fx/issues/357

chart.getRenderers().get(0).setAssumeSortedData(false)

Great work with chart-fx. Please keep going!

**Reply by wirew0rm** [reply]
Hey, thanks for sharing your problem and especially of course the solution you found :+1:
I remember that we thought about whether we should enforce sorting of the data in the dataset when we introduced that property, but essentially there's many (high throughput) use-cases where the data just can be assumed to be sorted which would have to pay the price for even just checking that the data is actually sorted. The same argument goes for disabling this by default. So basically the user has to know that this setting exists, which is not really ideal since it can be difficult to debug.
